simapp/app.go

```diff
@@ -227,6 +230,43 @@ func NewSimApp(
 	// NOTE: The testingkey is just mounted for testing purposes. Actual applications should
 	// not include this key.
 	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey, "testingkey")
+	// initialize stores
+	setNamespaces := func(config *baseapp.StoreParams, ver uint64) error {
+		for _, key := range keys {
+			typ, err := storev2.StoreKeyToType(key)
+			if err != nil {
+				return err
+			}
+			if err = config.RegisterSubstore(key.Name(), typ); err != nil {
+				return err
+			}
+		}
+		for _, key := range memKeys {
+			typ, err := storev2.StoreKeyToType(key)
+			if err != nil {
+				return err
+			}
+			if err = config.RegisterSubstore(key.Name(), typ); err != nil {
+				return err
+			}
+		}
+		for _, key := range tkeys {
+			typ, err := storev2.StoreKeyToType(key)
+			if err != nil {
+				return err
+			}
+			if err = config.RegisterSubstore(key.Name(), typ); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	baseAppOptions = append(baseAppOptions, baseapp.StoreOption(setNamespaces))
```

Comment:
This is way too much boilerplate to add to baseapp. We will need to find a better way. Realistically, I think it is unlikely that this will land in 0.46 and maybe we should time this with the app wiring and abci++ baseapp refactor